examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,216 @@
-# informative features
-X, y = datasets.make_classification(n_samples=100000, n_features=20,
-                                    n_informative=2, n_redundant=10,
-                                    random_state=42)
+from sklearn.metrics import precision_score, recall_score, f1_score
+from sklearn.calibration import CalibratedClassifierCV, plot_calibration_curve
+
+# %%
+# Dataset
+# -------
+#
+# We will use a synthetic binary classification dataset with 100,000 samples
+# and 20 features. Of the 20 features, only 2 are informative, 10 are
+# redundant (random combinations of the informative features) and the
+# remaining 8 are 'useless' (random numbers). Of the 100,000 samples, 1,000
+# will be used for model fitting and the rest for testing.
+
+X, y = make_classification(n_samples=100000, n_features=20, n_informative=2,
+                           n_redundant=10, random_state=42)
 
 X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.99,
                                                     random_state=42)
 
+# %%
+# Calibration curves
+# ------------------
+#
+# First, we will compare:
+#
+# * :class:`~sklearn.linear_model.LogisticRegression` (used as baseline
+#   since very often, properly regularized logistic regression is well
+#   calibrated by default thanks to the use of log-loss)
+# * Raw :class:`~sklearn.naive_bayes.GaussianNB`
+# * :class:`~sklearn.naive_bayes.GaussianNB` with isotonic and sigmoid
+#   calibration (see :ref:`User Guide <calibration>`)
+#
+# Calibration curves for all 4 conditions are plotted below, with the average
+# predicted probability for each bin on the x-axis and the fraction of positive
+# classes in each bin on the y-axis.
+#
+
+lr = LogisticRegression(C=1.)
+gnb = GaussianNB()
+gnb_isotonic = CalibratedClassifierCV(gnb, cv=2, method='isotonic')
+gnb_sigmoid = CalibratedClassifierCV(gnb, cv=2, method='sigmoid')
+
+clf_list = [(lr, 'Logistic'),
+            (gnb, 'Naive Bayes'),
+            (gnb_isotonic, 'Naive Bayes + Isotonic'),
+            (gnb_sigmoid, 'Naive Bayes + Sigmoid')]
+
+fig = plt.figure(figsize=(10, 10))
+ax1 = plt.subplot2grid((4, 2), (0, 0), rowspan=2, colspan=2)
+colors = plt.cm.get_cmap('Dark2')
+
+viz_objects = {}
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    viz = plot_calibration_curve(
+        clf, X_test, y_test, n_bins=10, name=name, ax=ax1, color=colors(i)
+    )
+    viz_objects[name] = viz
+
+ax1.grid()
+ax1.set_title('Calibration plots (Naive Bayes)')
+ax1.set(xlabel="")
+
+# Add histogram
+for i, (_, name) in enumerate(clf_list):
+    if i <= 1:
+        ax = plt.subplot2grid((4, 2), (2, i))
+    else:
+        ax = plt.subplot2grid((4, 2), (3, i - 2))
+
+    ax.hist(
+        viz_objects[name].y_prob, range=(0, 1), bins=10, label=name,
+        color=colors(i)
+    )
+    ax.set(title=name, xlabel="Mean predicted probability", ylabel="Count")
+
+plt.tight_layout()
```

Comment:
Based on the [matplotlib documentation](https://matplotlib.org/3.1.0/gallery/userdemo/demo_gridspec01.html#subplot2grid-demo), it is generally preferred to use the `GridSpec` api:

```py
from matplotlib.gridspec import GridSpec

fig = plt.figure(figsize=(10, 10))
gs = GridSpec(4, 2)
colors = plt.cm.get_cmap('Dark2')

ax1 = fig.add_subplot(gs[:2, :2])
viz_objects = {}
for i, (clf, name) in enumerate(clf_list):
    clf.fit(X_train, y_train)
    viz = plot_calibration_curve(
        clf, X_test, y_test, n_bins=10, name=name, ax=ax1, color=colors(i)
    )
    viz_objects[name] = viz

ax1.grid()
ax1.set_title('Calibration plots (Naive Bayes)')
ax1.set(xlabel="")

# Add histogram
grid_positions = [(2, 0), (2, 1), (3, 0), (3, 1)]
for i, (_, name) in enumerate(clf_list):
    row, col = grid_positions[i]
    ax = fig.add_subplot(gs[row, col])

    ax.hist(
        viz_objects[name].y_prob, range=(0, 1), bins=10, label=name,
        color=colors(i)
    )
    ax.set(title=name, xlabel="Mean predicted probability", ylabel="Count")
```

Reply:
Thanks! I should have seen that note!